app/src/androidTest/java/com/github/h3lp3rs/h3lp/HelperPageActivityTest.kt

```diff
@@ -143,6 +145,23 @@ class HelpPageActivityTest : H3lpAppTest() {
         }
     }
 
+    @Test
+    fun acceptingAnEmergencySendsPublicKeyOnDb() {
+        setupEmergencyAndDo {
+            // Accept
+            onView(withId(R.id.button_accept)).perform(click())
+
+            val conversationIds = databaseOf(CONVERSATION_IDS).getObjectsList(
+                TEST_EMERGENCY_ID,
+                Int::class.java
+            ).get()
+
+            // retrieve the public key sent on the database
+            val key = databaseOf(MESSAGES).getString(publicKeyPath(conversationIds[conversationIds.size - 1].toString(), HELPER.name)).get()
```

Comment:
You can use the thenAccept{ AssertNotNull....}. to allow other tests to run while waiting for the future to complete. Otherwise they are both correct so it's no big deal!